### PR TITLE
Add safe access to UrlReferrer in Log404

### PR DESCRIPTION
**Pull request by engineering87**
Allow safe access to the UrlReferrer property in Log404 in case of bad format UrlReferrer property.
Issue #3339

## Summary
We intercept the error in Log404 method and get the missing information from the header.

### CODE REVIEW

File: DNN Platform/Library/Entities/Urls/UrlRewriterUtils.cs [hunk 1]
```diff
@@ -111,9 +111,16 @@ public static void Log404(HttpRequest request, FriendlyUrlSettings settings, Url
...
-                if (request.UrlReferrer != null)
+                try
                 {
-                    log.LogProperties.Add(new LogDetailInfo("Referer", request.UrlReferrer.AbsoluteUri));
+                    if (request.UrlReferrer != null)
+                    {
+                        log.LogProperties.Add(new LogDetailInfo("Referer", request.UrlReferrer.AbsoluteUri));
+                    }
+                }
+                catch (Exception)
```
Comment (bdukes) [inline DNN Platform/Library/Entities/Urls/UrlRewriterUtils.cs]: Can this be changed to catch the specific exception type, please?

Comment (engineering87) [reply DNN Platform/Library/Entities/Urls/UrlRewriterUtils.cs]: Done

**Comment by bdukes** [comment]
This is targeting `development` which will be DNN 10, if you want this fix in DNN 9.5, we can retarget it to `release/9.4.x` (which is currently in code freeze for the 9.4.3 RC, so it we would wait to merge it in until 9.4.3 is officially out).

**Comment by engineering87** [comment]
Better in DNN 9.5 if it is possible.
Thank you

**Comment by bdukes** [comment]
I wasn't able to push my changes to your branch (maybe because you made the PR off of `development`, or maybe because you didn't check the "accept contributions from maintainers" checkbox), so I pushed it to a new PR, #3341

**Comment by engineering87** [comment]
For a mistake I merged another pull request related to another issue (https://github.com/dnnsoftware/Dnn.Platform/issues/2868) with this one, is it possible to split with another one? Thank you

**Comment by bdukes** [comment]
You'll want to create a new branch, rather than using `development`.  Assuming your change is a bug fix for DNN 9, rather than a feature/breaking change for DNN 10, here are some steps you should be able to follow:

First, make sure you have your remotes setup.  You'll want one for your fork on GitHub, and one for the main fork.  If you run `git remote --verbose` it will show you your remotes, and you can add one if you're missing.  It should look something like this:

```
origin  git@github.com:sharpcode-it/Dnn.Platform.git (fetch)
origin  git@github.com:sharpcode-it/Dnn.Platform.git (push)
upstream        https://github.com/dnnsoftware/Dnn.Platform.git (fetch)
upstream        https://github.com/dnnsoftware/Dnn.Platform.git (push)
```

So long as you have a remote pointing to a URL with `dnnsoftware` and one pointing to `sharpcode-it`, you're good (the URL can be in either format, probably `https`), just make sure you use the corresponding remote name below.  If you need to add one, you can use this command: `git remote add upstream https://github.com/dnnsoftware/Dnn.Platform.git`

```
git fetch --all
git checkout upstream/release/9.4.x
git checkout -b bug/update-module-settings
git cherry-pick 771570936e77ff397b46e09bcc8c9f24daa57ec0
git push --set-upstream origin bug/update-module-settings
```

1. Retrieve the latest commits from the remote repositories
2. Set your local repository to the contents of the `release/9.4.x` branch of the main repository.  You'll get a warning that you're not on a branch here, that's okay.  The next step is to create the branch.
3. Create a new branch named `bug/update-module-settings`
4. Copy the commit you created previously into this new branch
5. Push this new branch to your GitHub fork (and connect your local branch to the remote branch)

Hope it helps!

**Comment by valadas** [comment]
Closing this one in favor of #3341

**Pull request closed without merging**